deepfence_server/ingesters/malware_ingester.go

```diff
@@ -0,0 +1,181 @@
+package ingesters
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/deepfence/ThreatMapper/deepfence_utils/log"
+	"github.com/deepfence/ThreatMapper/deepfence_utils/utils"
+	"github.com/deepfence/ThreatMapper/deepfence_worker/ingesters"
+	"github.com/twmb/franz-go/pkg/kgo"
+
+	"github.com/deepfence/ThreatMapper/deepfence_utils/directory"
+)
+
+type MalwareIngester struct{}
+
+type Malware struct {
+	DocId               string `json:"doc_id"`
+	Timestamp           string `json:"@timestamp"`
+	Count               int    `json:"count,omitempty"`
+	Reason              string `json:"reason"`
+	Resource            string `json:"resource"`
+	Status              string `json:"status"`
+	Region              string `json:"region"`
+	AccountID           string `json:"account_id"`
+	Group               string `json:"group"`
+	Service             string `json:"service"`
+	Title               string `json:"title"`
+	ComplianceCheckType string `json:"compliance_check_type"`
+	CloudProvider       string `json:"cloud_provider"`
+	NodeName            string `json:"node_name"`
+	NodeID              string `json:"node_id"`
+	ScanID              string `json:"scan_id"`
+	Masked              string `json:"masked"`
+	Type                string `json:"type"`
+	ControlID           string `json:"control_id"`
+	Description         string `json:"description"`
+	Severity            string `json:"severity"`
+}
+
+func NewMalwareIngester() KafkaIngester[[]map[string]interface{}] {
+	return &MalwareIngester{}
+}
+
+func (tc *MalwareIngester) Ingest(
+	ctx context.Context,
+	cs []map[string]interface{},
+	ingestC chan *kgo.Record,
+) error {
+	tenantID, err := directory.ExtractNamespace(ctx)
+	if err != nil {
+		return err
+	}
+
+	rh := []kgo.RecordHeader{
+		{Key: "tenant_id", Value: []byte(tenantID)},
+	}
+
+	for _, c := range cs {
+		cb, err := json.Marshal(c)
+		if err != nil {
+			log.Error().Msg(err.Error())
+		} else {
+			ingestC <- &kgo.Record{
+				Topic:   utils.MALWARE_SCAN,
+				Value:   cb,
+				Headers: rh,
+			}
+		}
+	}
+
+	return nil
+}
+
+//func (tc *MalwareIngester) Ingest(ctx context.Context,
+//	cs []map[string]interface{},
+//	ingestC chan *kgo.Record) error {
+//	driver, err := directory.Neo4jClient(ctx)
+//	session, err := driver.Session(neo4j.AccessModeWrite)
+//
+//	rh := []kgo.RecordHeader{
+//		{Key: "tenant_id", Value: []byte(tenantID)},
+//	}
+//
+//	if err != nil {
+//		return err
+//	}
+//	defer session.Close()
+//
+//	tx, err := session.BeginTransaction()
+//	if err != nil {
+//		return err
+//	}
+//	defer tx.Close()
+//
+//	malwareMap := []map[string]interface{}{}
+//	for _, i := range cs {
+//		malware := map[string]interface{}{}
+//		match := i["StringsToMatch"].([]string)
+//		metaRules := i["MetaRules"].(map[string]string)
```

Comment:
Looks like in previous code we were actually extracting MetaRules
Can we follow the same path to avoid duplication across all malware results?